src/mlio/csv_reader.cxx

```diff
@@ -289,26 +289,40 @@ csv_reader::init_parsers_and_schema()
     auto col_beg = tbb::make_zip_iterator(idx_beg, name_beg, type_beg);
     auto col_end = tbb::make_zip_iterator(idx_end, name_end, type_end);
 
+    std::unordered_map<std::string, int> name_count;
+
     for (auto col_pos = col_beg; col_pos < col_end; ++col_pos) {
         std::size_t idx = std::get<0>(*col_pos);
-
-        std::string const &name = std::get<1>(*col_pos);
-
+        std::string name = std::get<1>(*col_pos);
         data_type dt = std::get<2>(*col_pos);
 
         if (should_skip(idx, name)) {
             skipped_columns_.emplace_back(1);
-
             column_parsers_.emplace_back(nullptr);
+            continue;
         }
-        else {
-            skipped_columns_.emplace_back(0);
 
-            column_parsers_.emplace_back(make_parser(dt, params_.parser_prm));
+        skipped_columns_.emplace_back(0);
+        column_parsers_.emplace_back(make_parser(dt, params_.parser_prm));
 
-            descs.emplace_back(
-                feature_desc_builder{name, dt, {batch_size, 1}}.build());
+        if (params_.dedupe_column_names) {
```

Comment:
The logic looks good to me. Below I used a couple "modern" C++ features to shorten the implementation and to reduce the number of lookup operations. I leave it up to you how much you want to modify your code:

```
if (params_.dedupe_column_names) {
    auto [pos, inserted] = name_count.try_emplace(name, 0);
    while (!inserted) {
        name.append("_").append(fmt::to_string(pos->second++));

        std::tie(pos, inserted) = name_count.try_emplace(name, 0);
    }
    pos->second++;
}
```

Reply:
Each new standard feels like a whole new language :) It took a while to understand this, but your suggestion is a lot better. I'm stealing it.